connectors/sources/sharepoint_online.py

```diff
@@ -1460,6 +1494,11 @@ async def get_docs_incrementally(self, sync_cursor, filtering=None):
                                 drive_item, access_control
                             )
 
+                            if self.configuration["fetch_drive_item_permissions"]:
```

Comment:
My personal preference would be moving this check into `_with_drive_item_permissions`, so you you don't have to check in multiple places.

Reply:
Adjusted with [e3820ba](https://github.com/elastic/connectors-python/pull/1259/commits/e3820babb0af8a8ad81da7b02cea0501568c298b)